FIAT/serendipity.py

```diff
@@ -130,7 +139,95 @@ def get_nodal_basis(self):
         raise NotImplementedError("get_nodal_basis not implemented for serendipity")
 
     def get_dual_set(self):
-        raise NotImplementedError("get_dual_set is not implemented for serendipity")
+        return self.dual
```

Comment:
I think a method like this should be among the inherited ones.

Furthermore, is this Serendipity a Ciarlet element now, or not?  If yes, we should change the base class...

Reply:
With your above comment, yes, it would be Ciarlet, but since we construct the dual from the nodal basis we should avoid calling the Ciarlet base class constructor.

Reply:
I looked at those dual constructions, and yes, the requirements are:
- reference cell
- degree
- `entity_ids`
- `tabulate` !!!

These are used on `self`.  And since tabulate, of course, needs a functioning element, but it is called inside the constructor... well, this looks very messy, but it is clear if this could be disentangled and how much...

Reply:
> but it is clear if this could be disentangled and how much...

Missing 'not', edited above.

Reply:
Yes, complicated.  It's not clear to me that, although we mathematically have a Ciarlet element, it makes sense to plumb it into that hierarchy.  Ciarlet elements take a given dual and a "wrong" basis and invert the vandermonde matrix to get the right nodal basis.  We already have tabulation in place and so don't need to re-construct it.  The lack of a nice dual seems to be a bigger limitation than not directly having a Ciarlet nodal basis.